rotkehlchen/chain/ethereum/graph.py

```diff
@@ -77,10 +84,42 @@ def query(
 
         querystr = prefix + querystr
         log.debug(f'Querying The Graph for {querystr}')
-        try:
-            result = self.client.execute(gql(querystr), variable_values=param_values)
-        except (requests.exceptions.RequestException, Exception) as e:
-            raise RemoteError(f'Failed to query the graph for {querystr} due to {str(e)}') from e
+
+        retries_left = QUERY_RETRY_TIMES
+        while retries_left > 0:
+            try:
+                result = self.client.execute(gql(querystr), variable_values=param_values)
+            except (requests.exceptions.RequestException, Exception) as e:
+                # NB: error status code expected to be in exception message
+                exc_msg = str(e)
+                is_retry_status_code = False
+                for status_code in RETRY_STATUS_CODES:
+                    if str(status_code.value) in exc_msg:
+                        is_retry_status_code = True
+                        break
+
+                if not is_retry_status_code:
+                    raise RemoteError(
+                        f'Failed to query the graph for {querystr} due to {str(e)}',
+                    ) from e
+
+                # Retry logic
+                retries_left -= 1
+                base_msg = (
+                    f'The Graph query to {querystr} failed due to the service is temporary down'
```

Comment:
> due to the service is temporary down

We can't know that. It depends on the error message, no? So perhaps just rename to

> `The Graph query to {querystr} failed due to {exc_msg}`